feat: Add documentation of MCP server and MCP Client possiblities

Matovidlo opened the pull request:
Jira issue(s): https://keboola.atlassian.net/browse/KAB-1086

<!-- briefly describe what are you changing and why -->

Changes:
- General overview of MCP integration and how to build data pipelines using natural language.
- Add section for MCP server.
- Add section for MCP client.
- Add section for MCP Local environment execution.

---

<!-- provide additional notes -->

Matovidlo commented:
@jordanrburger I have created documentation regarding MCP
1. On main page there is Building Data pipelines using natural language which leads to `integrations/mcp`
2. On `Keboola overview` page is short intro and again lead to `/integrations/mcp` page
3. The Integration page itself contains
3.1. Integration using existing Client which leads to `help.keboola.com/mcp` which does not exists ATM
3.2. Bring your own MCP client
3.3. Capabilities of MCP server (Here I am not sure)
3.4. Recommended and non recommended way how to run it locally and cursor IDE configuration connecting to MCP using docker/python

I hope it is quite alright

davidesner commented on integrate/mcp.md:
````diff
@@ -0,0 +1,197 @@
+
+4.  **Run the server (example using Python module directly for testing CLI manually)**:
+    ```bash
+    uv run python -m src.keboola_mcp_server.cli --api-url $KBC_API_URL
````
@Matovidlo `src` is not a python module, just a source folder the command should be `uv run python -m keboola_mcp_server`

Matovidlo commented on integrate/mcp.md:
````diff
@@ -0,0 +1,197 @@
+    }
+    ```
+    **Note:**
+    *   The exact command (`uvx`) and arguments might vary based on how `keboola_mcp_server` is installed and made available in your environment if you are pointing to a globally installed version. If using a local development version, the `command` and `args` would point to your local Python script and its execution, as described in the "Running Keboola MCP Server Locally" section.
````
5104a2e04b5e7091bae00eb6c8df995363125974

Matovidlo commented:
I have updated the docs @davidesner . I removed whole cloning part so no `python` dependencies except `uv`/`uvx` is required when we would like to run MCP server. I also removed `uv run` part 14de4abd90477fa43b3ac0e51b255cb5da001451

I maybe miss more promotion that it is open source and anyone can contribute? But I am not sure whether it should be there

davidesner commented:
> I have updated the docs @davidesner . I removed whole cloning part so no `python` dependencies except `uv`/`uvx` is required when we would like to run MCP server. I also removed `uv run` part [14de4ab](https://github.com/keboola/developers-docs/commit/14de4abd90477fa43b3ac0e51b255cb5da001451)
> 
> I maybe miss more promotion that it is open source and anyone can contribute? But I am not sure whether it should be there

Yes! That's a good point! We should mention this and especially call to action to star the Github repo and link to it.

davidesner commented:
There's one more thing that might be worth mentioning and that's the distinction between remote (Keboola hosted) version and local deployment (https://keboola.atlassian.net/browse/KAB-1088). And mention use cases.

- Use with IDE / AI Assistant (both [hosted](https://docs.google.com/document/d/12WkVb0NiLjwea4qcF2ObNdPfoPsIYYQDeOEwhvn_zn8/edit?tab=t.0) and local versions)
- Use as tooling for your agents -> [OpenAI SDK](https://openai.github.io/openai-agents-python/mcp/), LangChain

Matovidlo commented:
@jordanrburger @davidesner 
* Added top section for GH reference and star.
* Added AI agents integrations.

Regarding AI agents integrations I am not sure how in depth we should go to, I am mostly referencing to the Agentic frameworks themselves. If we have some examples of integrations which I currently do not have we can add it later on.

davidesner commented:
@Matovidlo @jordanrburger  it looks good to me. I made some updates and added deeplinks, please review
